internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -109,12 +114,52 @@ func createMonitoringStack(ctx context.Context, rr *odhtypes.ReconciliationReque
 	if !msExists {
 		return errors.New("MonitoringStack CRD not found")
 	}
+
 	if monitoring.Spec.Metrics != nil {
-		template := []odhtypes.TemplateInfo{
+		templates := []odhtypes.TemplateInfo{
 			{
 				FS:   resourcesFS,
 				Path: MonitoringStackTemplate,
 			},
+			{
+				FS:   resourcesFS,
+				Path: PrometheusRouteTemplate,
+			},
+		}
+
+		rr.Templates = append(rr.Templates, templates...)
+	}
+
+	return nil
+}
+
+func createOpenTelemetryCollector(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	otcExists, _ := cluster.HasCRD(ctx, rr.Client, gvk.OpenTelemetryCollector)
+	if !otcExists {
+		return errors.New("OpentelemetryCollector CRD not found")
```

Comment:
Not sure if you want to actually just return an error or setting a condition for better visibility ? so i.e. looking at the status you'd know if telemetry is enabled or not.